src/overlays/actors/ovl_En_Go2/z_en_go2.c

```diff
@@ -30,199 +216,2028 @@ const ActorInit En_Go2_InitVars = {
+    }
+    if (EnGo2_IsFreeingGoronInFire(this, globalCtx)) {
+        this->isAwake = false;
+        EnGo2_WakeUp(this, globalCtx);
+    }
+    if (((this->actor.params & 0x1F) != GORON_FIRE_GENERIC) && EnGo2_IsWakingUp(this)) {
+        EnGo2_WakeUp(this, globalCtx);
+    }
+}
+
+void func_80A46B40(EnGo2* this, GlobalContext* globalCtx) {
+    u8 index;
+    f32 height;
+
+    index = (this->actor.params & 0x1F);
+    if (this->unk_211 == true) {
+        EnGo2_BiggoronAnimation(this);
+        EnGo2_GoronLinkAnimation(this, globalCtx);
+        EnGo2_SelectGoronWakingUp(this);
+
+        if ((EnGo2_IsGoronRollingBig(this, globalCtx) == false) && (EnGo2_IsGoronFireGeneric(this) == false)) {
+            if (EnGo2_IsGoronDmtBombFlower(this)) {
+                return;
+            }
+        } else {
+            return;
+        }
+
+    } else {
+        if (func_800A56C8(&this->skelAnime, this->skelAnime.animFrameCount)) {
+            if ((this->actor.params & 0x1F) == GORON_DMT_BIGGORON) {
+                this->actor.flags |= 1;
+            }
+            func_80A454CC(this);
+            this->unk_211 = true;
+            this->collider.dim.height = (&D_80A4816C[index])->height;
+        } else {
+            height = (&D_80A4816C[index])->height;
+            this->collider.dim.height =
+                (s16)((height * 0.4f * (this->skelAnime.animCurrentFrame / this->skelAnime.animFrameCount)) +
+                      (height * 0.6f));
+        }
+    }
+    if ((EnGo2_IsCameraModified(this, globalCtx) == 0) && (EnGo2_IsWakingUp(this) == false)) {
+        EnGo2_RollingAnimation(this, globalCtx);
+    }
+}
+
+void EnGo2_GoronDmtBombFlowerAnimation(EnGo2* this, GlobalContext* globalCtx) {
+    f32 float1;
+    f32 float2;
+
+    float1 = this->skelAnime.animFrameCount;
+    float2 = this->skelAnime.animCurrentFrame * (32768.0f / float1);
+    this->actor.speedXZ = Math_Sins(float2);
+    if ((EnGo2_Orient(this, globalCtx)) && (this->waypoint == 0)) {
+        EnGo2_GetItemAnimation(this, globalCtx);
+    }
+}
+
+void EnGo2_GoronRollingBigContinueRolling(EnGo2* this, GlobalContext* globalCtx) {
+    if (func_800A56C8(&this->skelAnime, this->skelAnime.animFrameCount)) {
+        EnGo2_GetDustData(this, 1);
+        this->skelAnime.animPlaybackSpeed = 0.0f;
+        EnGo2_BeginRolling(this, globalCtx);
+    }
+}
+
+void EnGo2_ContinueRolling(EnGo2* this, GlobalContext* globalCtx) {
+    f32 float1 = 1000.0f;
+
+    if (((this->actor.params & 0x1F) != GORON_DMT_ROLLING_SMALL || !(this->actor.xyzDistFromLinkSq > SQ(float1))) &&
+        DECR(this->animTimer) == 0) {
+        this->actionFunc = EnGo2_SlowRolling;
+        this->actor.speedXZ *= 0.5f; // slowdown
+    }
+    EnGo2_GetDustData(this, 2);
+}
+
+void EnGo2_SlowRolling(EnGo2* this, GlobalContext* globalCtx) {
+    s32 orientation;
+    s32 index;
+
+    if (EnGo2_IsRolling(this) == false) {
+        if (EnGo2_IsRollingOnGround(this, 4, 8.0f, 1) == 1) {
```

Comment:
You named this function like a boolean so I would suggest:
```suggestion
        if (EnGo2_IsRollingOnGround(this, 4, 8.0f, 1)) {
```

Reply:
This suggestion should be non-equivalent, since `if` evaluates to true on non-zero, not one. If `EnGo2_IsRollingOnGround` returns a boolean state as the function name suggests, it should be this instead:

```suggestion
        if (EnGo2_IsRollingOnGround(this, 4, 8.0f, 1) == true) {
```